### ISSUE TITLE
[YSQL] Unexpected value of yb_get_effective_transaction_isolation_level() in java tests

### ISSUE THREAD

**Issue by pao214** [yugabyte/yugabyte-db#21711]
Jira Link: [DB-10589](https://yugabyte.atlassian.net/browse/DB-10589)
## Context

The following patch raises an assertion error

```
diff --git a/java/yb-pgsql/src/test/java/org/yb/pgsql/TestPgBatch.java b/java/yb-pgsql/src/test/java/org/yb/pgsql/TestPgBatch.java
index 1515b868a7..cb77b8c376 100644
--- a/java/yb-pgsql/src/test/java/org/yb/pgsql/TestPgBatch.java
+++ b/java/yb-pgsql/src/test/java/org/yb/pgsql/TestPgBatch.java
@@ -91,7 +91,9 @@ public class TestPgBatch extends BasePgSQLTest {
     Thread t = startConflictingThread(conflictingStatement, failureDetected, startSignal);
     try (Connection c = getConnectionBuilder().connect();
         Statement s = c.createStatement()) {
-      s.execute("BEGIN TRANSACTION ISOLATION LEVEL " + isolationLevel.sql);
+      s.execute("BEGIN");
+      //assertOneRow(s, "show yb_effective_transaction_isolation_level", "read committed");
+      assertOneRow(s, "SELECT yb_get_effective_transaction_isolation_level()", "read committed");
       for (int i = 1; i < 5; i++) {
         s.addBatch(String.format("UPDATE t SET v=2 WHERE k=%d", i));
       }
@@ -140,8 +142,8 @@ public class TestPgBatch extends BasePgSQLTest {
   public void testTransparentRestart() throws Throwable {
     // Params: conflicting statement, isolation level
     testTransparentRestartHelper(5, RC);
-    testTransparentRestartHelper(5, RR);
-    testTransparentRestartHelper(5, SR);
+    //testTransparentRestartHelper(5, RR);
+    //testTransparentRestartHelper(5, SR);
   }

   @Test
```

when built in debug (or fastdebug) mode.

CLI command used

```
./yb_build.sh debug --java-test 'org.yb.pgsql.TestPgBatch#testTransparentRestart' --clean-all --no-ccache

...

[ERROR] testTransparentRestart(org.yb.pgsql.TestPgBatch)  Time elapsed: 100.946 s  <<< FAILURE!
java.lang.AssertionError:
Column #1 mismatch: expected:<read committed> but was:<repeatable read>
Expected row: Row[0=java.lang.String::read committed]
Actual row:   Row[yb_get_effective_transaction_isolation_level=com.yugabyte.util.PGobject::repeatable read]
	at org.yb.pgsql.TestPgBatch.testTransparentRestartHelper(TestPgBatch.java:96)
	at org.yb.pgsql.TestPgBatch.testTransparentRestart(TestPgBatch.java:144)
```

## Issue

This behavior is unexpected since the command is supposed to return `read committed` in debug mode.

Moreover, should the line

```
flagMap.put("yb_enable_read_committed_isolation", "true");
```

ensure that the flag/function `yb_get_effective_transaction_isolation_level()` returns `read committed` and not `repeatable read`?

### Issue Type

kind/bug

### Warning: Please confirm that this issue does not contain any sensitive information

- [X] I confirm this issue does not contain any sensitive information.

[DB-10589]: https://yugabyte.atlassian.net/browse/DB-10589?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by pao214** [comment]
Landed the java test fix on master and backported to 2024.1

**Comment by pao214** [comment]
@pkj415 should we also backport this to 2.20 since 2.20 is LTS?
